ruby/ql/lib/codeql/ruby/security/UnsafeDeserializationCustomizations.qll

```diff
@@ -97,12 +98,15 @@ module UnsafeDeserialization {
 
   /**
    * An argument in a call to `YAML.parse*`, considered a sink for unsafe deserialization
-   * if there is a call to `to_ruby` on the returned value.
+   * if there is a call to `to_ruby` on the returned value of any Successor.
    */
   class YamlParseArgument extends Sink {
     YamlParseArgument() {
       this =
-        yamlNode().getAMethodCall(["parse", "parse_stream", "parse_file"]).getAMethodCall("to_ruby")
+        [
+          yamlParseChildNodeAccess(_).getMethod("to_ruby").getReturn().asSource(),
+          yamlParseChildNodeAccess(_).getMethod("to_ruby").getReturn().getAnElement().asSource()
```

Comment:
The sink should be the receiver of the `to_ruby` call. The alert location could possibly be improved in the query itself, but the sink should not be the return value of a call.